### PR TITLE
Automatically log PageRoute transitions

### PR DESCRIPTION
**Pull request by goderbauer**
Fixes https://github.com/flutter/flutter/issues/11216.
Requires https://github.com/flutter/flutter/pull/11236 (Travis failures are expected until it's submitted).

### CODE REVIEW

File: packages/firebase_analytics/lib/firebase_analytics.dart [hunk 1]
```diff
@@ -92,7 +92,7 @@ class FirebaseAnalytics {
   /// https://firebase.google.com/docs/reference/android/com/google/firebase/analytics/FirebaseAnalytics.html#setCurrentScreen(android.app.Activity, java.lang.String, java.lang.String)
   /// https://firebase.google.com/docs/reference/ios/firebaseanalytics/api/reference/Classes/FIRAnalytics#setscreennamescreenclass
   Future<Null> setCurrentScreen(
-      {@required String screenName, String screenClassOverride}) async {
+      {@required String screenName, String screenClassOverride: 'Flutter'}) async {
```
Comment (collinjackson) [inline packages/firebase_analytics/lib/firebase_analytics.dart]: Small nit that I also saw in another PR today: is it Flutter style to put spaces on the inside of curly braces? I realize that this file is formatted without these spaces so feel free to leave it as is for consistency if you don't want to reformat the file.

File: packages/firebase_analytics/example/lib/tabs_page.dart [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+  }
+
+  @override
+  void didPopNext() {
+    _sendCurrentTabToAnalytics();
+  }
+
+  void _sendCurrentTabToAnalytics() {
+    observer.analytics.setCurrentScreen(
+        screenName: '${TabsPage.routeName}/tab$selectedIndex');
```
Comment (collinjackson) [inline packages/firebase_analytics/example/lib/tabs_page.dart]: I would probably use a trailing comma here so you don't have to double-indent it

**Comment by collinjackson** [comment]
Looks like travis is complaining about a missing import

File: packages/firebase_analytics/lib/firebase_analytics.dart (hunk 1 above)
Comment (goderbauer) [inline packages/firebase_analytics/lib/firebase_analytics.dart]: For the flutter/plugins repo we're using the dart formatter (enforced by Travis), which doesn't put the spaces there.

File: packages/firebase_analytics/example/lib/tabs_page.dart (hunk 2 above)
Comment (goderbauer) [inline packages/firebase_analytics/example/lib/tabs_page.dart]: Done.

**Pull request merged**